backend/inspirehep/records/api/mixins.py

```diff
@@ -69,6 +69,15 @@ def citations_by_year(self):
         db_query = db_query.group_by("year").order_by("year")
         return [{"year": r.year.year, "count": r.sum} for r in db_query.all() if r.year]
 
+    def hard_delete(self):
+        with db.session.begin_nested():
+            LOGGER.warning("Hard Deleting citations")
+            # Removing citations from RecordCitations table
+            RecordCitations.query.filter_by(citer_id=self.id).delete()
+            # Removing references to this record from RecordCitations table
+            RecordCitations.query.filter_by(cited_id=self.id).delete()
```

Comment:
Here we could do something like `RecordCitations.query.filter_by(or_(citer_id=self.id, cited_id=self.id))`, WDYT?